File: 2021/day-1/typescript/solve-puzzle-2.ts
import { readFileSync } from 'node:fs';

const input: string[] = readFileSync('../input.txt', 'utf-8').split('\n');
const len: number = input.length;
const nums: number[] = [];

let timesIncreased: number = 0;

for (
  let i: number = 0, first: boolean = true, previousSum: number | null = null;
  i < len;
  i++
) {
  if (first) {
    first = false;

    nums.push(Number(input[i]), Number(input[i + 1]), Number(input[i + 2]));

    previousSum = nums[0] + nums[1] + nums[2];
    continue;
  }

  nums.push(Number(input[i + 2]));

  const currentSum: number = nums[i] + nums[i + 1] + nums[i + 2];

  if (currentSum > previousSum) timesIncreased++;

  previousSum = currentSum;
}

console.log(timesIncreased);

File: 2021/day-1/python/solve-puzzle-2.py
with open('../input.txt', 'r') as handle:
    text = handle.read().splitlines()
    nums = [ int(text[0]), int(text[1]), int(text[2]) ]
    previous_sum = sum(nums)
    times_increased = 0
    limit = len(text) - 2
    i = 1
    #print(limit)
    while i < limit:
        nums.append(int(text[i + 2]))
        #print(i, end=">")
   
        current_sum = nums[i] + nums[i + 1] + nums[i + 2]
        if current_sum > previous_sum:
            #print(len(nums), end=" ")
            times_increased += 1
        
        previous_sum = current_sum
        i += 1
    #print()
    print(times_increased)